### PR TITLE
Allow changing Task States Colors

### LINKED ISSUE #7843: Colorblind friendly UI
Hello! Thanks for all your work on Airflow :) 

### Description

Reds (failed) & greens (success) colors are not colorblind friendly. 

### Use case / motivation

As a deuteranope, when I view recently run dags, it's not clear which runs succeeded and which failed. To me, this is probably one of the most important features of the Airflow UI. 

Screen Shot 2020-03-23 at 10 18 59 AM

*You could convince me everything succeeded or failed in this screenshot.*

### Suggestions

Totally understand that red/green are important for our common expectations of failure/success, so I'll just propose some other reds/greens that work for me, with the caveat that there are many more types of colorblindness than my own so it's worth getting a diverse set of opinions here. Something other than default `red` and `green` web colors will be tremendously helpful though!

**`red` / `DarkSeaGreen`** - simple change, works well with the yellow "running" value

Screen Shot 2020-03-23 at 10 21 58 AM

**[clrs.cc](http://clrs.cc/)** failed = `#FF4136`, success = `#2ECC40`, running = `#01FF70` - this comes with the bonus of being [A11Y compliant](http://clrs.cc/a11y/).

Screen Shot 2020-03-23 at 10 31 45 AM

Happy to create more combinations, if helpful. Cheers!

### PR DESCRIPTION
**Pull request by kaxil**
closes https://github.com/apache/airflow/issues/7843

The main purpose of this PR is to allow **Colorblind Friendly UI**

**Stock Screenshots**:

image

image
image
image
image
image
image

**Steps to change colors**:

Create `airflow_local_settings.py` file and put in on PYTHONPATH. (example put that file under `$AIRFLOW_HOME/config/airflow_local_settings.py` with the following contents:

```python
STATE_COLORS = {
    "queued": 'darkgray',
    "running": '#01FF70',
    "success": '#2ECC40',
    "failed": 'firebrick',
    "up_for_retry": 'yellow',
    "up_for_reschedule": 'turquoise',
    "upstream_failed": 'orange',
    "skipped": 'darkorchid',
    "scheduled": 'tan',
}
```

**Screenshots with the above changes**:

image
image
image
image
image
image

---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Comment by kaxil** [comment]
Once the PR is approved, I will add some docs on changing colors

**Comment by Acehaidrey** [comment]
Hi @kaxil thanks for your work on this. Quick question from your notes I'm not seeing how to make a colorblind friendly UI here though as a user configuration. If we just have local_configuration, is the idea that we just have this configurable for the webserver for everyone. I thought it was to make it configurable at a per person level.

**Comment by kaxil** [comment]
> Hi @kaxil thanks for your work on this. Quick question from your notes I'm not seeing how to make a colorblind friendly UI here though as a user configuration. If we just have local_configuration, is the idea that we just have this configurable for the webserver for everyone. I thought it was to make it configurable at a per person level.

Hey, yes this PR won't allow per-user change. It would only allow per deployment change. I wanted to address custom colors per deployment for 1.10.11 and we can take a look at per-user change at a later stage (maybe for 2.0)

**Review by Acehaidrey — APPROVED** [review]

**Comment by Acehaidrey** [comment]
I see makes sense. The colors were hardcoded up to this point and weren't customizable at all - just want to make sure that my understanding is correct.

File: airflow/www/views.py [hunk 1]
```diff
@@ -336,6 +337,7 @@ def get_int_arg(value, default=0):
 
         state_color_mapping = State.state_color.copy()
         state_color_mapping["null"] = state_color_mapping.pop(None)
+        state_color_mapping.update(STATE_COLORS)
```
Comment (ashb) [inline airflow/www/views.py]: Isn't this already done from https://github.com/apache/airflow/pull/9520/files#diff-e198cdb4d2a7b4801616a067e6cbd82eR80

Comment (kaxil) [reply airflow/www/views.py]: Updated

Comment (kaxil) [reply airflow/www/views.py]: in https://github.com/apache/airflow/pull/9520/commits/f9601c21dade06bd737af95ed09c5bd946c3b722

**Comment by kaxil** [comment]
Rebased on latest master

**Comment by ashb** [comment]
Happy with this now @kaxil?

**Comment by kaxil** [comment]
> Happy with this now @kaxil?

Yup

**Review by ashb — APPROVED** [review]

**Comment by preston-m-price** [comment]
Thanks for this change!

**Pull request merged**